File: src/main/java/io/junseok/mealmate/domain/memberwish/entity/MemberWish.java
package io.junseok.mealmate.domain.memberwish.entity;

import io.junseok.mealmate.domain.board.entity.Board;
import io.junseok.mealmate.domain.member.entity.Member;
import io.junseok.mealmate.domain.restaurant.entity.Restaurant;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "member_wish")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Builder
public class MemberWish {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_wish_id")
    private Long memberWishId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    private Member member;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;
}

File: src/main/java/io/junseok/mealmate/domain/memberwish/entity/MemberWish.kt
package io.junseok.mealmate.domain.memberwish.entity

import io.junseok.mealmate.domain.member.entity.Member
import io.junseok.mealmate.domain.restaurant.entity.Restaurant
import javax.persistence.*

@Entity
@Table(name = "member_wish")
class MemberWish(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_wish_id")
    var memberWishId: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    val member: Member,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "restaurant_id")
    val restaurant: Restaurant
)
